uisdk/src/test/java/tech/dojo/pay/uisdk/presentation/ui/carddetailscheckout/viewmodel/CardDetailsCheckoutViewModelTest.kt

```diff
@@ -92,830 +92,878 @@ class CardDetailsCheckoutViewModelTest {
+            )
+            // act
+            val viewModel = CardDetailsCheckoutViewModel(
+                observePaymentIntent,
+                dojoCardPaymentHandler,
+                observePaymentStatus,
+                updatePaymentStateUseCase,
+                getSupportedCountriesUseCase,
+                supportedCountriesViewEntityMapper,
+                allowedPaymentMethodsViewEntityMapper,
+                cardCheckoutScreenValidator,
+                cardCheckOutFullCardPaymentPayloadMapper,
+                stringProvider,
+                isStartDestination,
+                refreshPaymentIntentUseCase,
+                getRefreshedPaymentTokenFlow,
+                navigateToCardResult,
+            )
+            // assert
+            Assert.assertEquals(expected, viewModel.state.value)
+        }
 
     @Test
-    fun `when payment intent flow collect with collect billing address false viewModel should emit state without billing country `() = runTest {
-        // arrange
-        val paymentIntentFakeFlow: MutableStateFlow<PaymentIntentResult?> = MutableStateFlow(null)
-        given(observePaymentIntent.observePaymentIntent()).willReturn(paymentIntentFakeFlow)
-        val paymentStateFakeFlow: MutableStateFlow<Boolean> = MutableStateFlow(true)
-        val supportedIcons = listOf(1, 2, 3)
-        given(observePaymentStatus.observePaymentStates()).willReturn(paymentStateFakeFlow)
-        paymentIntentFakeFlow.tryEmit(
-            PaymentIntentResult.Success(
-                result = PaymentIntentDomainEntity(
-                    "id",
-                    "token",
-                    AmountDomainEntity(
-                        10L,
-                        "100",
-                        "GBP",
+    fun `when payment intent flow collect with collect billing address false viewModel should emit state without billing country `() =
+        runTest {
+            // arrange
+            val paymentIntentFakeFlow: MutableStateFlow<PaymentIntentResult?> =
+                MutableStateFlow(null)
+            given(observePaymentIntent.observePaymentIntent()).willReturn(paymentIntentFakeFlow)
+            val paymentStateFakeFlow: MutableStateFlow<Boolean> = MutableStateFlow(true)
+            val supportedIcons = listOf(1, 2, 3)
+            given(observePaymentStatus.observePaymentStates()).willReturn(paymentStateFakeFlow)
+            paymentIntentFakeFlow.tryEmit(
+                PaymentIntentResult.Success(
+                    result = PaymentIntentDomainEntity(
+                        "id",
+                        "token",
+                        AmountDomainEntity(
+                            10L,
+                            "100",
+                            "GBP",
+                        ),
+                        supportedCardsSchemes = listOf(CardsSchemes.AMEX),
                     ),
-                    supportedCardsSchemes = listOf(CardsSchemes.AMEX),
                 ),
-            ),
```

Comment:
instead of emitting you can just return the flow with the value you want on mock. That's better than creating a mutable flow and emitting the value
```kotlin
val paymentResult = PaymentIntentResult.Success(
    result = PaymentIntentDomainEntity(
        "id",
        "token",
        AmountDomainEntity(
            10L,
            "100",
            "GBP",
        ),
        supportedCardsSchemes = listOf(CardsSchemes.AMEX),
    ),
),
given(observePaymentIntent.observePaymentIntent()).willReturn(flowOf(paymentResult))
```

Reply:
this will be part of the flow reactor that will happen later  as now the use case is returning mutable flow